### DISCUSSION TITLE
Add own DIY sensor

### DISCUSSION THREAD

**Discussion by BansheeLemming** [Rem0o/FanControl.Releases#884] [Q&A]
Hi,

I own a custom water loop computer without any temperature connectors on the Mainboard. (But two installed temperature sensors)
And since I am cheap and do not want to buy for example a 50 bucks aquacomputer sensor reader my idea was to use an Arduino to read out my sensors (10k NTC IIRC) and have it send data to my system via a USB interface. (before you may ask - doing the Arduino part is not an issue at all. E.g. I have a temperature readout that makes logical sense.)

What I dont know (and sadly dont know how to look for) is what kind of protocols and or data formats are nessecary to display my DIY setup to this tool.
Maybe I need to write an interface that converts data to be used in this tool? 

Is it at all possible to do what I plan to do? Any help is much appreciated!
If this question was already posted somewhere, please link me to that thread. My own search for this was not succesful.

**Comment by Rem0o** [comment]
There are two ways.

1) You can write a plugin that import your custom sensor, see plugin section in readme.
2) You can use a file sensor and have your program write to that file the raw number as a string.
